// element-plus/lib/components/radio/index2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../utils/index2.js');
var radio$1 = require('./src/radio4.js');
var radioButton$1 = require('./src/radio-button4.js');
var radioGroup$1 = require('./src/radio-group4.js');
var radio = require('./src/radio3.js');
var radioGroup = require('./src/radio-group3.js');
var radioButton = require('./src/radio-button3.js');
var install = require('../../utils/vue/install2.js');

const ElRadio = install.withInstall(radio$1["default"], {
  RadioButton: radioButton$1["default"],
  RadioGroup: radioGroup$1["default"]
});
const ElRadioGroup = install.withNoopInstall(radioGroup$1["default"]);
const ElRadioButton = install.withNoopInstall(radioButton$1["default"]);

exports.radioEmits = radio.radioEmits;
exports.radioProps = radio.radioProps;
exports.radioPropsBase = radio.radioPropsBase;
exports.useRadio = radio.useRadio;
exports.radioGroupEmits = radioGroup.radioGroupEmits;
exports.radioGroupProps = radioGroup.radioGroupProps;
exports.radioButtonProps = radioButton.radioButtonProps;
exports.ElRadio = ElRadio;
exports.ElRadioButton = ElRadioButton;
exports.ElRadioGroup = ElRadioGroup;
exports["default"] = ElRadio;
