// element-plus/es/components/popconfirm/src/popconfirm3.mjs
import '../../button/index2.mjs';
import { QuestionFilled } from '@element-plus/icons-vue';
import '../../../utils/index2.mjs';
import '../../tooltip/index2.mjs';
import { buttonTypes } from '../../button/src/button3.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props2.mjs';
import { iconPropType } from '../../../utils/vue/icon2.mjs';
import { useTooltipContentProps } from '../../tooltip/src/tooltip3.mjs';

const popConfirmButtonTypes = [...buttonTypes, "text"];
const popconfirmProps = buildProps({
  title: String,
  confirmButtonText: String,
  cancelButtonText: String,
  confirmButtonType: {
    type: String,
    values: popConfirmButtonTypes,
    default: "primary"
  },
  cancelButtonType: {
    type: String,
    values: popConfirmButtonTypes,
    default: "text"
  },
  icon: {
    type: iconPropType,
    default: QuestionFilled
  },
  iconColor: {
    type: String,
    default: "#f90"
  },
  hideIcon: {
    type: Boolean,
    default: false
  },
  hideAfter: {
    type: Number,
    default: 200
  },
  onConfirm: {
    type: definePropType(Function)
  },
  onCancel: {
    type: definePropType(Function)
  },
  teleported: useTooltipContentProps.teleported,
  persistent: useTooltipContentProps.persistent
});

export { popconfirmProps };
