// element-plus/es/components/table-v2/src/grid2.mjs
import '../../../utils/index2.mjs';
import '../../virtual-list/index2.mjs';
import { columns, dataType, fixedDataType, requiredNumber, classType, styleType } from './common2.mjs';
import { tableV2HeaderProps } from './header2.mjs';
import { tableV2RowProps } from './row2.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props2.mjs';
import { virtualizedListProps, virtualizedGridProps } from '../../virtual-list/src/props2.mjs';

const tableV2GridProps = buildProps({
  columns,
  data: dataType,
  fixedData: fixedDataType,
  estimatedRowHeight: tableV2RowProps.estimatedRowHeight,
  width: requiredNumber,
  height: requiredNumber,
  headerWidth: requiredNumber,
  headerHeight: tableV2HeaderProps.headerHeight,
  bodyWidth: requiredNumber,
  rowHeight: requiredNumber,
  cache: virtualizedListProps.cache,
  useIsScrolling: Boolean,
  scrollbarAlwaysOn: virtualizedGridProps.scrollbarAlwaysOn,
  scrollbarStartGap: virtualizedGridProps.scrollbarStartGap,
  scrollbarEndGap: virtualizedGridProps.scrollbarEndGap,
  class: classType,
  style: styleType,
  containerStyle: styleType,
  getRowHeight: {
    type: definePropType(Function),
    required: true
  },
  rowKey: tableV2RowProps.rowKey,
  onRowsRendered: {
    type: definePropType(Function)
  },
  onScroll: {
    type: definePropType(Function)
  }
});

export { tableV2GridProps };
