// element-plus/es/components/table-v2/src/row2.mjs
import '../../../utils/index2.mjs';
import '../../virtual-list/index2.mjs';
import { columns, expandColumnKey, rowKey } from './common2.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props2.mjs';
import { virtualizedGridProps } from '../../virtual-list/src/props2.mjs';

const tableV2RowProps = buildProps({
  class: String,
  columns,
  columnsStyles: {
    type: definePropType(Object),
    required: true
  },
  depth: Number,
  expandColumnKey,
  estimatedRowHeight: {
    ...virtualizedGridProps.estimatedRowHeight,
    default: void 0
  },
  isScrolling: Boolean,
  onRowExpand: {
    type: definePropType(Function)
  },
  onRowHover: {
    type: definePropType(Function)
  },
  onRowHeightChange: {
    type: definePropType(Function)
  },
  rowData: {
    type: definePropType(Object),
    required: true
  },
  rowEventHandlers: {
    type: definePropType(Object)
  },
  rowIndex: {
    type: Number,
    required: true
  },
  rowKey,
  style: {
    type: definePropType(Object)
  }
});

export { tableV2RowProps };
