// element-plus/lib/components/virtual-list/src/defaults.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defaults = require('./defaults2.js');



exports.AUTO_ALIGNMENT = defaults.AUTO_ALIGNMENT;
exports.BACKWARD = defaults.BACKWARD;
exports.CENTERED_ALIGNMENT = defaults.CENTERED_ALIGNMENT;
exports.DEFAULT_DYNAMIC_LIST_ITEM_SIZE = defaults.DEFAULT_DYNAMIC_LIST_ITEM_SIZE;
exports.END_ALIGNMENT = defaults.END_ALIGNMENT;
exports.FORWARD = defaults.FORWARD;
exports.HORIZONTAL = defaults.HORIZONTAL;
exports.ITEM_RENDER_EVT = defaults.ITEM_RENDER_EVT;
exports.LTR = defaults.LTR;
exports.PageKey = defaults.PageKey;
exports.RTL = defaults.RTL;
exports.RTL_OFFSET_NAG = defaults.RTL_OFFSET_NAG;
exports.RTL_OFFSET_POS_ASC = defaults.RTL_OFFSET_POS_ASC;
exports.RTL_OFFSET_POS_DESC = defaults.RTL_OFFSET_POS_DESC;
exports.SCROLLBAR_MIN_SIZE = defaults.SCROLLBAR_MIN_SIZE;
exports.SCROLL_EVT = defaults.SCROLL_EVT;
exports.SMART_ALIGNMENT = defaults.SMART_ALIGNMENT;
exports.START_ALIGNMENT = defaults.START_ALIGNMENT;
exports.ScrollbarDirKey = defaults.ScrollbarDirKey;
exports.ScrollbarSizeKey = defaults.ScrollbarSizeKey;
exports.VERTICAL = defaults.VERTICAL;
