// element-plus/lib/utils/index2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('./dom/index2.js');
require('./vue/index2.js');
var animation = require('./animation2.js');
var arrays = require('./arrays2.js');
var browser = require('./browser2.js');
var error = require('./error2.js');
require('./functions2.js');
var i18n = require('./i18n2.js');
var objects = require('./objects2.js');
var raf = require('./raf2.js');
var rand = require('./rand2.js');
var strings = require('./strings2.js');
var types = require('./types2.js');
var typescript = require('./typescript2.js');



exports.cubic = animation.cubic;
exports.easeInOutCubic = animation.easeInOutCubic;
exports.castArray = arrays.castArray;
exports.unique = arrays.unique;
exports.isFirefox = browser.isFirefox;
exports.debugWarn = error.debugWarn;
exports.throwError = error.throwError;
exports.isKorean = i18n.isKorean;
exports.entriesOf = objects.entriesOf;
exports.getProp = objects.getProp;
exports.keysOf = objects.keysOf;
exports.cAF = raf.cAF;
exports.rAF = raf.rAF;
exports.generateId = rand.generateId;
exports.getRandomInt = rand.getRandomInt;
exports.escapeStringRegexp = strings.escapeStringRegexp;
exports.isElement = types.isElement;
exports.isEmpty = types.isEmpty;
exports.isPropAbsent = types.isPropAbsent;
exports.isUndefined = types.isUndefined;
exports.mutable = typescript.mutable;
