// element-plus/es/locales.mjs
import './locales2.mjs';
export { default as af } from './locale/lang/af2.mjs';
export { default as ar } from './locale/lang/ar2.mjs';
export { default as az } from './locale/lang/az2.mjs';
export { default as bg } from './locale/lang/bg2.mjs';
export { default as bn } from './locale/lang/bn2.mjs';
export { default as ca } from './locale/lang/ca2.mjs';
export { default as cs } from './locale/lang/cs2.mjs';
export { default as da } from './locale/lang/da2.mjs';
export { default as de } from './locale/lang/de2.mjs';
export { default as el } from './locale/lang/el2.mjs';
export { default as en } from './locale/lang/en2.mjs';
export { default as eo } from './locale/lang/eo2.mjs';
export { default as es } from './locale/lang/es2.mjs';
export { default as et } from './locale/lang/et2.mjs';
export { default as eu } from './locale/lang/eu2.mjs';
export { default as fa } from './locale/lang/fa2.mjs';
export { default as fi } from './locale/lang/fi2.mjs';
export { default as fr } from './locale/lang/fr2.mjs';
export { default as he } from './locale/lang/he2.mjs';
export { default as hr } from './locale/lang/hr2.mjs';
export { default as hu } from './locale/lang/hu2.mjs';
export { default as hyAm } from './locale/lang/hy-am2.mjs';
export { default as id } from './locale/lang/id2.mjs';
export { default as it } from './locale/lang/it2.mjs';
export { default as ja } from './locale/lang/ja2.mjs';
export { default as kk } from './locale/lang/kk2.mjs';
export { default as km } from './locale/lang/km2.mjs';
export { default as ko } from './locale/lang/ko2.mjs';
export { default as ku } from './locale/lang/ku2.mjs';
export { default as ky } from './locale/lang/ky2.mjs';
export { default as lt } from './locale/lang/lt2.mjs';
export { default as lv } from './locale/lang/lv2.mjs';
export { default as mn } from './locale/lang/mn2.mjs';
export { default as nbNo } from './locale/lang/nb-no2.mjs';
export { default as nl } from './locale/lang/nl2.mjs';
export { default as pa } from './locale/lang/pa2.mjs';
export { default as pl } from './locale/lang/pl2.mjs';
export { default as ptBr } from './locale/lang/pt-br2.mjs';
export { default as pt } from './locale/lang/pt2.mjs';
export { default as ro } from './locale/lang/ro2.mjs';
export { default as ru } from './locale/lang/ru2.mjs';
export { default as sk } from './locale/lang/sk2.mjs';
export { default as sl } from './locale/lang/sl2.mjs';
export { default as sr } from './locale/lang/sr2.mjs';
export { default as sv } from './locale/lang/sv2.mjs';
export { default as ta } from './locale/lang/ta2.mjs';
export { default as th } from './locale/lang/th2.mjs';
export { default as tk } from './locale/lang/tk2.mjs';
export { default as tr } from './locale/lang/tr2.mjs';
export { default as ugCn } from './locale/lang/ug-cn2.mjs';
export { default as uk } from './locale/lang/uk2.mjs';
export { default as uzUz } from './locale/lang/uz-uz2.mjs';
export { default as vi } from './locale/lang/vi2.mjs';
export { default as zhCn } from './locale/lang/zh-cn2.mjs';
export { default as zhTw } from './locale/lang/zh-tw2.mjs';
