// element-plus/lib/tokens/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('./index2.js');
var breadcrumb = require('./breadcrumb2.js');
var button = require('./button2.js');
var carousel = require('./carousel2.js');
var collapse = require('./collapse2.js');
var configProvider = require('./config-provider2.js');
var dialog = require('./dialog2.js');
var form = require('./form2.js');
var pagination = require('./pagination2.js');
var radio = require('./radio2.js');
var row = require('./row2.js');
var scrollbar = require('./scrollbar2.js');
var tabs = require('./tabs2.js');
var upload = require('./upload2.js');
var popper = require('./popper2.js');
var tooltipV2 = require('./tooltip-v22.js');



exports.breadcrumbKey = breadcrumb.breadcrumbKey;
exports.buttonGroupContextKey = button.buttonGroupContextKey;
exports.carouselContextKey = carousel.carouselContextKey;
exports.collapseContextKey = collapse.collapseContextKey;
exports.configProviderContextKey = configProvider.configProviderContextKey;
exports.dialogInjectionKey = dialog.dialogInjectionKey;
exports.formContextKey = form.formContextKey;
exports.formItemContextKey = form.formItemContextKey;
exports.elPaginationKey = pagination.elPaginationKey;
exports.radioGroupKey = radio.radioGroupKey;
exports.rowContextKey = row.rowContextKey;
exports.scrollbarContextKey = scrollbar.scrollbarContextKey;
exports.tabsRootContextKey = tabs.tabsRootContextKey;
exports.uploadContextKey = upload.uploadContextKey;
exports.POPPER_CONTENT_INJECTION_KEY = popper.POPPER_CONTENT_INJECTION_KEY;
exports.POPPER_INJECTION_KEY = popper.POPPER_INJECTION_KEY;
exports.TOOLTIP_V2_OPEN = tooltipV2.TOOLTIP_V2_OPEN;
exports.tooltipV2ContentKey = tooltipV2.tooltipV2ContentKey;
exports.tooltipV2RootKey = tooltipV2.tooltipV2RootKey;
